rsocket/statemachine/RSocketStateMachine.h

```diff
@@ -58,6 +59,7 @@ class FrameSink {
 class RSocketStateMachine final
     : public FrameSink,
       public FrameProcessor,
+      public ManageableConnection,
```

Comment:
Can we have RSocketStateMachine compose, rather than inherit, from another class? It's already difficult to tell what `override` methods correspond to what base class.

Reply:
I did not add any functions to RSocketStateMachine and actually removed a hack-y function. I will take a look but it might be another improvement on top of this. I want to keep my changes limited.

Reply:
(y) fair enough